app-modules/prospect/src/Actions/UpdateProspect.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/*
+<COPYRIGHT>
+
+    Copyright © 2016-2025, Canyon GBS LLC. All rights reserved.
+
+    Advising App™ is licensed under the Elastic License 2.0. For more details,
+    see https://github.com/canyongbs/advisingapp/blob/main/LICENSE.
+
+    Notice:
+
+    - You may not provide the software to third parties as a hosted or managed
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace AdvisingApp\Prospect\Actions;
+
+use AdvisingApp\Prospect\DataTransferObjects\UpdateProspectData;
+use AdvisingApp\Prospect\Models\Prospect;
+use AdvisingApp\Prospect\Models\ProspectSource;
+use AdvisingApp\Prospect\Models\ProspectStatus;
+use Illuminate\Support\Str;
+use Illuminate\Validation\ValidationException;
+
+class UpdateProspect
+{
+    public function execute(Prospect $prospect, UpdateProspectData $data): Prospect
+    {
+        $prospect->fill($data->toArray());
+
+        if ($data->status) {
+            $status = ProspectStatus::whereRaw('LOWER(name) = ?', [Str::lower($data->status)])->first();
+            throw_if(
+                ! $status,
+                ValidationException::withMessages(['status' => 'Status does not exist.'])
+            );
+            $prospect->status_id = $status->getKey();
+        }
+
+        if ($data->source) {
+            $source = ProspectSource::whereRaw('LOWER(name) = ?', [Str::lower($data->source)])->first();
+            throw_if(
+                ! $source,
+                ValidationException::withMessages(['source' => 'Source does not exist.'])
+            );
+            $prospect->source_id = $source->getKey();
+        }
```

Comment:
Same as for the create, I would do this validation in the controller and have the data object have the models themselves stored in it.

And set the relationships using the relationship methods.